app/renderer/components/tabs/tabPage.js

```diff
@@ -44,21 +50,20 @@ class TabPage extends ImmutableComponent {
   }
 
   onDrop (e) {
-    if (this.props.tabPageFrames.size === 0) {
+    if (this.props.isPageEmpty) {
       return
     }
 
     appActions.dataDropped(getCurrentWindowId())
 
-    const moveToFrame = this.props.tabPageFrames.get(0)
     const sourceDragData = dndData.getDragData(e.dataTransfer, dragTypes.TAB)
     const sourceDragFromPageIndex = this.props.sourceDragFromPageIndex
     // This must be executed async because the state change that this causes
     // will cause the onDragEnd to never run
     setTimeout(() => {
       // If we're moving to a right page, then we're already shifting everything to the left by one, so we want
       // to drop it on the right.
-      windowActions.moveTab(sourceDragData.get('key'), moveToFrame.get('key'),
+      windowActions.moveTab(sourceDragData.get('key'), this.props.moveToFrameKey,
```

Comment:
Should we still be using windowActions / frameKey? I think it would make more sense to use appAction / tabId